### ISSUE TITLE
Have you considered using sass + css modules as the main technical point

### ISSUE THREAD

**Issue by MinJieLiu** [zeit/next.js#1690]
`with-external-scoped-css` is not perfect. 
In a slightly larger system, doing so will add a lot of work.
 I think `style jsx` is not a good idea, it can not and other open source UI framework very good integration. 

Have you considered using `sass` + `css modules` as the main technical point

**Comment by code4aman** [comment]
Imo, css-in-js is better for component styling. And even more, with tools like PreJSS you can write both - (any steroid) css and css-in-js.
Anyway, there is a problem with plain css for components styling - duplicates in `next/head`. With this fix released you can write your own wrapper to reduce boilerplate code for styles content injection. Take a look at my example [here](https://github.com/zeit/next.js/pull/668#issuecomment-292797221)

**Comment by khankuan** [comment]
@code4aman thanks for your tip! I have a similar problem. I'm using a ui kit that is built on top of css modules. I would like to use the ui kit in my next.js application. It seems like next.js does not support css modules though it isn't global. So practically, i'm thinking of exporting the ui kit into a .js and .css file that I'll use in the application, I'll need to live with the fact that the ui kit has to be imported globally for now. Your solution is great but does it mean that the entire css would be downloaded on every page load? I'm thinking of copying the css file into the static folder (during build process somehow) and then import it as a head `<link ref='stylesheet'>` tag pointing to the static file. What do you think?

Cheers! :)

**Comment by code4aman** [comment]
@khankuan that's a default behavior and yes, this way your styles will be referenced on every page load. The real problem is, since we are component-oriented, do we reference only styles the current page uses? To import them on-demand, in per-component manner, we need that PR. And we need to somehow (my example wrapper is just a point to start with) reduce current boilerplate code for css injection.
If your 3rd party framework provides one large stylesheets file then your only way is to reference it in `_document.js` or maybe your own layout file.

**Comment by khankuan** [comment]
@code4aman would your method allow browser chacing?

**Comment by code4aman** [comment]
@khankuan yes, why not. CSS Modules will change hash only if content is changed, so it's safe to rebuild.

**Comment by khankuan** [comment]
Oh a related note, did you manage to get CSS modules working? I have a uikit that is based on css modules + sass + react. I tried overriding the webpack config but didn't manage to get it working. I then tried building my uikit into a umd lib with a css file. But I realized that it didn't work well with styled-jsx because it does not pass `data-jsx` prop to components, so I could not do layout styling in my application. Any help/advice would be appreciated :)
